pkg/k8shandler/cluster.go

```diff
@@ -47,7 +47,13 @@ func (er *ElasticsearchRequest) CreateOrUpdateElasticsearchCluster() error {
 	}
 	wrongConfig = false
 
+	// Populate nodes from the custom resources spec.nodes and update the
+	// cluster status immediately to refresh status.nodes before progressing
+	// with any unschedulable or new nodes.
 	er.populateNodes()
+	if err := er.UpdateClusterStatus(); err != nil {
+		return err
+	}
```

Comment:
nit, but based on your description it would make sense to me that this goes right before the progressing of unschedulable nodes rather than being after we populateNodes.

or maybe we can push this into the progressUnschedulableNodes function and make it the first operation within there

Reply:
Sure, I would say it is better to keep this right before calling `progressUnschedulableNodes` instead of pulling it in. Pulling it would try to update twice the status, even if the first status update fails.